radicle-node/src/service/message.rs

```diff
@@ -497,9 +503,30 @@ impl ZeroBytes {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use quickcheck_macros::quickcheck;
+    use crate::prelude::*;
+    use crate::wire::Encode;
 
     use crate::crypto::test::signer::MockSigner;
+    use crate::test::arbitrary;
+    use quickcheck_macros::quickcheck;
+
+    #[test]
+    fn test_inventory_limit() {
+        let msg = Message::inventory(
+            InventoryAnnouncement {
+                inventory: arbitrary::vec(INVENTORY_LIMIT)
+                    .try_into()
+                    .expect("size within bounds limit"),
+                timestamp: LocalTime::now().as_secs(),
+            },
+            &MockSigner::default(),
+        );
+        let mut buf: Vec<u8> = Vec::new();
+        assert!(
+            msg.encode(&mut buf).is_ok(),
+            "INVENTORY_LIMIT is too big to support message encoding",
+        );
```

Comment:
Can you add an assert for decoding too?

Reply:
Sure, though does the property test for encoding/decoding cover this already?

https://github.com/radicle-dev/heartwood/blob/master/radicle-node/src/wire/message.rs#L405

Reply:
In fact, the property test should cover both cases already right?

Reply:
It doesn't (see unwrap) -- it only produces valid messages.

Reply:
I went ahead and implemented it while waiting on the answer.

Reply:
I must be missing something, I don't see where this is tested?

Reply:
Indeed, its annoying.  Clicking on the file link of the comment points to a specific commit.   I thought I forgot to push the changes.  Maybe because the comment was on the specific commit?

Here are the relevant changes: https://github.com/radicle-dev/heartwood/pull/115/commits/0f7543799bd9c333344e20091a96bb3456638f3b

Reply:
I saw those changes, but they don't test that decoding fails with a too big payload.

Reply:
```rust
        assert!(
            msg.encode(&mut buf).is_ok(),
            "INVENTORY_LIMIT is too big to support message encoding",
        );
```
The message here also doesn't make sense, it should say `"the inventory size is smaller than the limit"`